src/wp-includes/class-wp-dependencies.php

```diff
@@ -199,10 +199,31 @@ public function all_deps( $handles, $recursion = false, $group = false ) {
 				continue;
 			}
 
-			$keep_going = true;
+			$keep_going           = true;
+			$missing_dependencies = array();
+			if ( isset( $this->registered[ $handle ] ) && $this->registered[ $handle ]->deps ) {
+				$missing_dependencies = array_diff( $this->registered[ $handle ]->deps, array_keys( $this->registered ) );
+			}
 			if ( ! isset( $this->registered[ $handle ] ) ) {
 				$keep_going = false; // Item doesn't exist.
-			} elseif ( $this->registered[ $handle ]->deps && array_diff( $this->registered[ $handle ]->deps, array_keys( $this->registered ) ) ) {
+			} elseif ( $missing_dependencies ) {
+				// Prevent duplicate notices.
+				static $reported = array();
+
+				if ( ! isset( $reported[ $handle ] ) ) {
+					$reported[ $handle ] = true;
+
+					_doing_it_wrong(
+						__METHOD__,
+						sprintf(
+							/* translators: 1: Script module ID, 2: Comma-separated list of missing dependency IDs. */
+							__( 'The script with the handle %1$s was enqueued with dependencies that are not registered: %2$s.' ),
```

Comment:
Problem: The changes here are in `WP_Dependencies` which is the base class for _both_ `WP_Scripts` _and_ `WP_Styles`. This means that this could very well be reporting issues for styles that have missing dependencies. So the translation string here would 

```suggestion
							/* translators: 1: Handle, 2: Comma-separated list of missing dependency handles. */
							__( 'The script with the handle %1$s was enqueued with dependencies that are not registered: %2$s.' ),
```

Also, the translators comment here is updated to for non-module scripts.

Reply:
Since this runs in WP_Dependencies, we need a message that works for both scripts and styles. One option is to determine the item type dynamically via get_class( $this ) (or instanceof) so the message outputs “script” or “style” as needed. Another option is to make the message more generic, for example:

`The handle %1$s was enqueued with dependencies that are not registered: %2$s.`

Reply:
What if we had a new `protected` method for getting the error message? In `WP_Dependencies` it can be generic and maybe include the class name. Then in `WP_Scipts` and `WP_Styles` the method can be overridden to have script- and style-specific messages? The method should take the handle and missing dependency handles as arguments.

Reply:
Good idea — I’ve added a protected `get_dependency_warning_message( $handle, $missing_dependencies )` method to `WP_Dependencies` and overridden it in both `WP_Scripts` and `WP_Styles`.